### ISSUE TITLE
What's in a name?

### ISSUE THREAD

**Issue by crozone** [PomeloFoundation/Lolita#22]
I'm not usually one to have issue with names of packages, but... wewlad.

I'm looking to use this package in a professional environment in combination with Pomelo, and the fact that it's called [Lolita](https://en.wikipedia.org/wiki/Lolita) makes things a touch awkward. I mean, I like Nabokov as much as the next guy, but _yikes_. I'm just trying to do some bulk updates without writing raw SQL, not get into a discussion with my boss and co-workers about the nuances of package names and their relation to a hugely influential and controversial work of 1950s erotic fiction.

Is there any practical scope to reconsider the naming of this package?

**Comment by nuclearpidgeon** [comment]
Yeah the name struck me as very strange as well. I know the book/movie is a somewhat well-known classic (I mean, I only noticed the name because I'd heard of the Stanley Kubrick movie), but... what on earth do bulk database operations have to do with "a middle-aged literature professor [...] obsessed with a 12-year-old girl"?

**Comment by lauxjpn** [comment]
I absolutely agree. Is this project still active or can we just deprecate it?

In case it is still active, I propose we change the name and then first deprecated the older packages on nuget.org and later hide them completely.

/cc @yukozh

**Comment by lauxjpn** [comment]
@yukozh Can this package/project **please** be renamed this month?

I pitch the name `Bulk`, but pretty much any name is an improvement over the current one.

**Comment by lauxjpn** [comment]
@yukozh Any update on the progress so far?

**Comment by lauxjpn** [comment]
@yukozh Please rename the project and package until **2023-11-30**. Otherwise, I will take the necessary actions myself without any further warning or dialog about it.

To make it perfectly clear, why we (PomeloFoundation) do not want to be associated with a project/package name like this one, here the first paragraph of the Wikipedia article about the book with the same title:

> _Lolita_ is a 1955 novel written by [Russian-American](https://en.wikipedia.org/wiki/Russian_Americans) novelist [Vladimir Nabokov](https://en.wikipedia.org/wiki/Vladimir_Nabokov) which addresses [hebephilia](https://en.wikipedia.org/wiki/Hebephilia). The protagonist is a French literature professor who moves to New England and writes under the pseudonym Humbert Humbert. He describes his obsession with a 12-year-old "[nymphet](https://en.wikipedia.org/wiki/Nymphet)", Dolores Haze, whom he kidnaps and [sexually abuses](https://en.wikipedia.org/wiki/Child_sexual_abuse) after becoming her stepfather.

It is also highly unlikely that anyone wants to ship a package of this project with the current name to a customer.

/cc @bgrainger

**Comment by bgrainger** [comment]
Other than a handful of commits from Nov-Jan, it looked like this project was completely inactive. One possible way forward would be to just archive the project, but that still leaves the name associated with "PomeloFoundation".

It looks like `Lolita` is used in public API names, so changing that would be a breaking change. For a package that only has small 10,000s of downloads, it hardly seems worth it. (Plus it looks like it could be a lot of work to go through and search-and-replace all instances of the names, while ensuring the code still compiles, etc.)

I wonder if it might be simplest to:

1. Replace the README with a link to this issue or an explanation of why the package is archived,
2. Archive the repo,
3. Unlist NuGet packages (if possible; you may not be an owner with permission to do that),
4. Allow anyone who wants to maintain this code to fork the repo and take ownership (or transfer the repo to @yukozh).

This doesn't remove the association of the name with PomeloFoundation but (a) that might not even be possible without contacting NuGet to get the packages taken down, and (b) it makes it clear that this package is not endorsed without requiring a lot of renaming work.
